SharpPcap/LibPcap/PcapDevice.cs

```diff
@@ -300,25 +285,28 @@ public virtual int GetNextPacket(out RawCapture p)
                 throw new InvalidOperationDuringBackgroundCaptureException("GetNextPacket() invalid during background capture");
             }
 
-            p = null;
-
             if (!PollFileDescriptor())
             {
+                e = default;
+
                 // We checked, there is no data using poll()
-                return 0;
+                return -1;
```

Comment:
Are you sure about this change?

Reply:
I think so. We were using pcap_next_ex() which returns 1 when it reads a packet, but this is a little atypical for function returns and WinDivertDevice.SendPacketArrivalEvent() was returning 0 upon success, non-zero upon failure. I figured we might as well standardize on returning 0 upon success like most functions do, but I'm open to other ideas like keeping it at 1 and standardizing on that, although that feels a little atypical (even though pcap and WinDivert return 1 upon success when receiving, maybe WinDivert copied pcap?), or using an enum etc.

Reply:
pcap_next_ex() returns 1 if the packet was read without problems, 0 if packets are being read from a live capture and the packet buffer timeout expired or a negative number on error.
if poll tell us thereis nothing to read, we should return 0 (timeout) and not -1 (error)

Reply:
Whatever we choose we should probably be consistent. Zero is typically 'good' with nonzero being not good. With the out parameter there isn't a good way to report whether we got or didn't get a packet via the parameter. Agree it doesn't match pcap_next_ex() but the user also doesn't know or care that pcap_next_ex() is what we are calling. And we are using the same return value between PcapDevice and WinDivertDevice.

Would you prefer 1 upon success and 0 upon no packet? I can adjust PcapDevice and WinDivertDevice to match in either case.

Reply:
Well to put it simply, Libpcap logic goes something like this, (for most functions, not just pcap_next_ex)
* Negative: Error
* Positive: Success
* Zero: Try again

Reply:
@kayoub5 it doesn't matter too much to me. Fixed to make it 0 on no data, 1 on data, and negative on error. Thoughts on using an enum to make the return value a bit more clear to the end user? I've already screwed it up a few times. Maybe there is some way to return the enum such that values not in the enum are able to be handled, like our negative case?